WordPress/gutenberg
# [RNMobile] Add 'Insert from URL' option to Image block

## What?
Fixes [wordpress-mobile/gutenberg-mobile#3190](https://github.com/wordpress-mobile/gutenberg-mobile/issues/3190) by adding "Insert from URL" option to the Image block. 

Related: [gutenberg-mobile#4779](https://github.com/wordpress-mobile/gutenberg-mobile/pull/4779)

## Why?
"Insert from URL" is an option for the Audio, Video, and Image blocks on the web, and also for the Audio block on Gutenberg Mobile. This PR adds the same web behavior to the Image block.

## How?
- Adds an onSelectURL handler to the native image block based on the [equivalent code](https://github.com/WordPress/gutenberg/blob/d5915916abc45e6682f4bdb70888aa41e98aa395/packages/block-library/src/image/edit.js#L268-L278) on from the web Image block.
- Adds error message handler for invalid URLs inspired by the same UI as the [native File block](https://github.com/WordPress/gutenberg/blob/d5915916abc45e6682f4bdb70888aa41e98aa395/packages/block-library/src/file/edit.native.js#L490-L503).
- Adds `MEDIA_TYPE_IMAGE` to the list of source options available when selecting an Image block, which is controlled by the Media Upload component.
- Updates the Media Upload test cases to be more explicit about the media upload options for each block. (More info about test strategies and other considerations forthcoming below.) 

## Testing Instructions

1. Open a Post or Page
2. Insert an Image Block
3. Tap on Insert from URL in the menu options
4. Paste a valid (or invalid) URL into the text input tap Apply
5. Image should display in UI (or show error message if the URL is invalid) 

https://user-images.githubusercontent.com/643285/165879097-dfbb5d57-d44e-4fe0-98cf-648dffa6a325.mov
